nilearn/_utils/niimg_conversions.py

```diff
@@ -394,3 +394,54 @@ def _index_niimgs(niimgs, index):
     return nibabel.Nifti1Image(niimgs.get_data()[:, :, :, index],
                                niimgs.get_affine(),
                                header=niimgs.get_header())
+
+
+def is_hdr_equal(niimg1, niimg2, skip_affine=True, skip_datatype=False):
+    """Compare two Niimg-like objects regarding their Nifti headers.
+
+    Parameters
+    ----------
+    niimg1/niimg2: Niimg-like object
+        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+        Specifies the two images whose headers will be compared. These can have
+        3D or 4D shape.
+
+    skip_affine: boolean
+       If True, the header fields related to the affine matrix (srow_x/y/z)
+       will be ignored. Use case: compare resampled images.
+
+    skip_datatype: boolean
+       If True, the header fields related to datatype of the image values
+       will be ignored. Use case: type conversion occurred.
+
+    Returns
+    -------
+    is_hdr_equal: boolean
+        True if niimg1 and niimg2 have identical header fields.
+    """
+    img1 = check_niimgs(niimg1, accept_3d=True)
+    img2 = check_niimgs(niimg2, accept_3d=True)
+
+    # create list of header fields to verify
+    from nibabel.nifti1 import header_dtd
+    hdr_items_to_check = [item[0] for item in header_dtd]
```

Comment:
Would it be possible to have this list explicitly, at least in a comment, or in a notes section in the docstring. That way one can immediately see what is compared.

For reference ```[('sizeof_hdr', 'i4'),
 ('data_type', 'S10'),
 ('db_name', 'S18'),
 ('extents', 'i4'),
 ('session_error', 'i2'),
 ('regular', 'S1'),
 ('dim_info', 'u1'),
 ('dim', 'i2', (8,)),
 ('intent_p1', 'f4'),
 ('intent_p2', 'f4'),
 ('intent_p3', 'f4'),
 ('intent_code', 'i2'),
 ('datatype', 'i2'),
 ('bitpix', 'i2'),
 ('slice_start', 'i2'),
 ('pixdim', 'f4', (8,)),
 ('vox_offset', 'f4'),
 ('scl_slope', 'f4'),
 ('scl_inter', 'f4'),
 ('slice_end', 'i2'),
 ('slice_code', 'u1'),
 ('xyzt_units', 'u1'),
 ('cal_max', 'f4'),
 ('cal_min', 'f4'),
 ('slice_duration', 'f4'),
 ('toffset', 'f4'),
 ('glmax', 'i4'),
 ('glmin', 'i4'),
 ('descrip', 'S80'),
 ('aux_file', 'S24'),
 ('qform_code', 'i2'),
 ('sform_code', 'i2'),
 ('quatern_b', 'f4'),
 ('quatern_c', 'f4'),
 ('quatern_d', 'f4'),
 ('qoffset_x', 'f4'),
 ('qoffset_y', 'f4'),
 ('qoffset_z', 'f4'),
 ('srow_x', 'f4', (4,)),
 ('srow_y', 'f4', (4,)),
 ('srow_z', 'f4', (4,)),
 ('intent_name', 'S16'),
 ('magic', 'S4')]
```